Source/RapyutaSimulationPlugins/Private/Robots/RRBaseRobot.cpp

```diff
@@ -28,9 +29,37 @@ void ARRBaseRobot::SetupDefault()
     // Generally, for sake of dynamic robot type import/creation, child components would be then created on the fly!
     // Besides, a default subobject, upon content changes, also makes the owning actor become vulnerable since one in child BP actor
     // classes will automatically get invalidated.
+    URRUObjectUtils::SetupDefaultRootComponent(this);
 
-    RootComponent = CreateDefaultSubobject<USceneComponent>(TEXT("Root"));
-    ROS2InterfaceClass = URRRobotROS2Interface::StaticClass();
+    // NOTE: Any custom object class (eg ROS2InterfaceClass) that is required to be configurable by this class' child BP ones
+    // & IF its object needs to be created before BeginPlay(),
+    // -> They must be left NULL here, so its object (eg ROS2Interface) is not created by default in [PostInitializeComponents()]
+}
+
+void ARRBaseRobot::PostInitializeComponents()
+{
+    Super::PostInitializeComponents();
+    if (ROS2InterfaceClass)
+    {
+        CreateROS2Interface();
+    }
+    else
+    {
+        // [OnConstruction] could run in various Editor BP actions, thus could not do Fatal log here
+        UE_LOG(LogRapyutaCore,
+               Warning,
+               TEXT("[%s] [ARRBaseRobot::PostInitializeComponents()] ROS2InterfaceClass has not been configured, "
+                    "probably later in child BP class!"),
+               *GetName());
+    }
+}
+
+void ARRBaseRobot::CreateROS2Interface()
+{
+    ROS2Interface = CastChecked<URRRobotROS2Interface>(

```

Comment:
Can the `CastChecked` make an exception and crash the editor if `ROS2InterfaceClass` is not initialized properly?